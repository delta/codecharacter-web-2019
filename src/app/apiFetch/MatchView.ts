/* tslint:disable:no-console*/
import { API_BASE_URL } from '../../config/config';

export const getMatches = () => {
  return fetch(`${API_BASE_URL}match/all`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};

export const getTopMatches = () => {
  return fetch(`${API_BASE_URL}match/pro`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};

export const getGameLogs = (gameId: number) => {
  return fetch(`${API_BASE_URL}match/log/${gameId}`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};
