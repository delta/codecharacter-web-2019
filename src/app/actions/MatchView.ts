import { Match } from 'app/types/MatchView';
import { action } from 'typesafe-actions';

export namespace MatchActions {
  export enum Type {
    GET_MATCHES = 'GET_MATCHES',
    GET_TOP_MATCHES = 'GET_TOP_MATCHES',
    VIEW_MATCH = 'VIEW_MATCH',
    UPDATE_LOADING_STATUS = 'UPDATE_LOADING_STATUS',
    UPDATE_ERROR = 'UPDATE_ERROR',
    UPDATE_MATCHES = 'UPDATE_MATCHES',
    UPDATE_TOP_MATCHES = 'UPDATE_TOP_MATCHES',
  }

  export const getMatches = () => action(Type.GET_MATCHES);

  export const getTopMatches = () => action(Type.GET_TOP_MATCHES);

  export const updateLoadingStatus = (loadingStatus: boolean) => {
    return action(Type.UPDATE_LOADING_STATUS, { loadingStatus });
  };

  export const updateMatches = (matchesData: Match[]) => {
    return action(Type.UPDATE_MATCHES, { matchesData });
  };

  export const updateTopMatches = (topMatchesData: Match[]) => {
    return action(Type.UPDATE_TOP_MATCHES, { topMatchesData });
  };

  export const updateError = (err: string) => {
    return action(Type.UPDATE_ERROR, err);
  };
}
