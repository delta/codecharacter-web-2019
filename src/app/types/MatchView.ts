import { MatchActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

export enum MatchViewTabType {
  MY_MATCHES = 'MY_MATCHES',
  TOP_MATCHES = 'TOP_MATCHES',
}

export interface OwnState {
  activeMatchViewTab: MatchViewTabType;
  offset: number;
}

export interface Match {
  userOne: string;
  userOneScore: number;
  userTwoScore: number;
  userTwo: string;
  winner: string;
  playedAt: Date;
}

export interface StateProps {
  matches: Match[];
  topMatches: Match[];
  loading: boolean;
}

export interface DispatchProps {
  getMatches: () => void;
  getTopMatches: () => void;
}

export type Props = StateProps & DispatchProps;

export type State = OwnState;

export interface ElementOwnProps {
  match: Match;
  index: number;
}

export type ElementProps = ElementOwnProps;

const actions = {
  getMatches: MatchActions.getMatches,
  getTopMatches: MatchActions.getTopMatches,
  updateError: MatchActions.updateError,
  updateLoading: MatchActions.updateLoadingStatus,
  updateMatches: MatchActions.updateMatches,
  updateTopMatchs: MatchActions.updateTopMatches,
};

export interface MatchStoreState {
  loading: boolean;
  matches: Match[];
  topMatches: Match[];
}

export type MatchStoreAction = ActionType<typeof actions>;
