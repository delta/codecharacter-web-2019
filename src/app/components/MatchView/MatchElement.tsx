import { faShieldAlt, faVideo } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as styles from 'app/styles/MatchView.module.css';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col } from 'react-bootstrap';

export class MatchElement extends React.Component<MatchInterfaces.ElementProps, {}> {
  public render() {
    const { match } = this.props;
    return (
      <Col sm={12} className={classnames('mb-1')}>
        <div
          className={classnames('w-100', styles.matchElement, {
            [`${styles.winBackground}`]: match.userOneScore > match.userTwoScore,
            [`${styles.loseBackground}`]: match.userOneScore < match.userTwoScore,
          })}
        >
          <div className="d-flex justify-content-around text-dark">
            <span>{match.userOne}</span>
            <span>
              <FontAwesomeIcon icon={faShieldAlt} />
            </span>
            <span>{match.userTwo}</span>
          </div>
          <div className=" d-flex justify-content-center text-capitalize text-font-weight-bold text-dark h2  my-2">
            {match.userOneScore} - {match.userTwoScore}
          </div>
          <div className=" d-flex justify-content-center text-capitalize text-font-weight-bold text-dark">
            {match.userOne === match.winner ? 'WON' : 'LOST'}
          </div>
          <div className={classnames(styles.body, 'd-flex justify-content-around text-dark')}>
            <span className={classnames(styles.game)}>
              <FontAwesomeIcon icon={faVideo} />
            </span>
            <span className={classnames(styles.game)}>
              <FontAwesomeIcon icon={faVideo} />
            </span>
            <span className={classnames(styles.game)}>
              <FontAwesomeIcon icon={faVideo} />
            </span>
            <span className={classnames(styles.game)}>
              <FontAwesomeIcon icon={faVideo} />
            </span>
            <span className={classnames(styles.game)}>
              <FontAwesomeIcon icon={faVideo} />
            </span>
          </div>
          <div
            className=" d-flex justify-content-start text-capitalize text-dark my-1 mx-3 h6"
            style={{
              fontSize: '12px',
            }}
          >
            Played At : {match.playedAt}
          </div>
        </div>
      </Col>
    );
  }
}
