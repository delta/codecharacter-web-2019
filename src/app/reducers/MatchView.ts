import { MatchActions } from 'app/actions';
import * as MatchViewInterfaces from 'app/types/MatchView';

const matchesInitialState: MatchViewInterfaces.MatchStoreState = {
  loading: false,
  matches: [],
  topMatches: [],
};

export const matchesReducer = (
  state = matchesInitialState,
  action: MatchViewInterfaces.MatchStoreAction,
) => {
  switch (action.type) {
    case MatchActions.Type.UPDATE_LOADING_STATUS: {
      return {
        ...state,
        loading: action.payload.loadingStatus,
      };
    }
    case MatchActions.Type.UPDATE_MATCHES: {
      return {
        ...state,
        loading: false,
        matches: action.payload.matchesData,
      };
    }
    case MatchActions.Type.UPDATE_TOP_MATCHES: {
      return {
        ...state,
        loading: false,
        topMatches: action.payload.topMatchesData,
      };
    }
    case MatchActions.Type.UPDATE_ERROR:
      return {
        ...state,
        loading: false,
      };
    case MatchActions.Type.RESET_MATCH_VIEW:
      return {
        ...matchesInitialState,
      };
    default:
      return state;
  }
};
