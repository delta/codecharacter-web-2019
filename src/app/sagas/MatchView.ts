import { GameLogActions, MatchActions } from 'app/actions';
import * as MatchFetch from 'app/apiFetch/MatchView';
import { checkAuthentication } from 'app/sagas/utils';
import { resType } from 'app/types/sagas';
import { all, call, put, takeEvery } from 'redux-saga/effects';
import { ActionType } from 'typesafe-actions';

export function* getMatches(action: ActionType<typeof MatchActions.getMatches>) {
  try {
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(MatchFetch.getMatches);
    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(MatchActions.updateMatches(res.matchData));
    }
  } catch (err) {
    throw err;
  }
}

export function* getTopMatches(action: ActionType<typeof MatchActions.getTopMatches>) {
  try {
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(MatchFetch.getTopMatches);

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(MatchActions.updateTopMatches(res.matchData));
    }
  } catch (err) {
    throw err;
  }
}

export function* getGameLogs(action: ActionType<typeof MatchActions.getGameLogs>) {
  try {
    const res = yield call(MatchFetch.getGameLogs, action.payload.gameId);

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(GameLogActions.clearDisplayDebugLog());

      const logs = res.logs;
      const debugLog1 = logs.player1Log;
      const debugLog2 = logs.player2Log;
      const gameLog = logs.gameLog;
      const matchPlayerId = logs.matchPlayerId;

      yield put(GameLogActions.updateGameLog('', '', ''));
      yield put(GameLogActions.updateGameLog(debugLog1, debugLog2, gameLog));
      yield put(GameLogActions.updateMatchPlayerId(matchPlayerId));
    }
  } catch (err) {
    throw err;
  }
}

export function* matchSagas() {
  yield all([
    takeEvery(MatchActions.Type.GET_MATCHES, getMatches),
    takeEvery(MatchActions.Type.GET_TOP_MATCHES, getTopMatches),
    takeEvery(MatchActions.Type.GET_GAME_LOGS, getGameLogs),
  ]);
}
