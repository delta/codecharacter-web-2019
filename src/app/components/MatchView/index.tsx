import { faCaretLeft, faCaretRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { MatchElement } from 'app/components/MatchView/MatchElement';
import * as styles from 'app/styles/MatchView.module.css';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Grid, Row } from 'react-bootstrap';
// tslint:disable-next-line
import ReactPaginate from 'react-paginate';

export class Match extends React.Component<MatchInterfaces.Props, MatchInterfaces.State> {
  private static paginationSize = 15;
  constructor(props: MatchInterfaces.Props) {
    super(props);
    this.state = {
      activeMatchViewTab: MatchInterfaces.MatchViewTabType.MY_MATCHES,
      offset: 0,
    };
  }

  public componentDidMount() {
    this.props.getMatches();
    this.props.getTopMatches();
  }

  public handlePageClick = (data: { selected: number }) => {
    this.setState({
      offset: Math.ceil(data.selected * Match.paginationSize),
    });
  };

  public render() {
    const { activeMatchViewTab } = this.state;
    const { matches, topMatches } = this.props;

    return (
      <Grid fluid={true} className={classnames(styles.MatchView)}>
        <Row className="justify-content-between py-2 px-3">
          <Col className="text-light font-weight-bold my-auto">MATCHES</Col>
        </Row>
        <Row>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.MY_MATCHES);
                this.props.getMatches();
              }}
            >
              {' '}
              My Matches{' '}
            </button>
          </div>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.TOP_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.TOP_MATCHES);
                this.props.getTopMatches();
              }}
            >
              {' '}
              Top Matches{' '}
            </button>
          </div>
        </Row>
        <Row className={classnames('mb-2', styles.matchViewWrap)}>
          {activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES ? (
            matches && matches.length ? (
              matches.map((match, index) =>
                match ? (
                  index >= this.state.offset &&
                  index <= this.state.offset + Match.paginationSize - 1 ? (
                    <MatchElement match={match} index={index - 1} key={index - 1} />
                  ) : null
                ) : (
                  <div className="ml-5"> Nothing to show </div>
                ),
              )
            ) : (
              <div className="ml-5"> Nothing to show </div>
            )
          ) : topMatches && topMatches.length ? (
            topMatches.map((match, index) =>
              match ? (
                index >= this.state.offset &&
                index <= this.state.offset + Match.paginationSize - 1 ? (
                  <MatchElement match={match} index={index - 1} key={index - 1} />
                ) : null
              ) : (
                <div className="ml-5"> Nothing to show </div>
              ),
            )
          ) : (
            <div className="ml-5"> Nothing to show </div>
          )}
        </Row>
        <Row>
          <Col className="d-flex justify-content-center" style={{ width: '100vw', margin: '10px' }}>
            <ReactPaginate
              previousLabel={
                <span>
                  <FontAwesomeIcon icon={faCaretLeft} /> <FontAwesomeIcon icon={faCaretLeft} />
                </span>
              }
              nextLabel={
                <span>
                  <FontAwesomeIcon icon={faCaretRight} /> <FontAwesomeIcon icon={faCaretRight} />
                </span>
              }
              breakLabel={'...'}
              breakClassName={'break-me'}
              marginPagesDisplayed={1}
              pageClassName={'atag'}
              pageRangeDisplayed={2}
              pageCount={Math.ceil(matches.length / Match.paginationSize)}
              activeLinkClassName={'active'}
              onPageChange={this.handlePageClick}
              containerClassName={'pagination'}
              activeClassName={'active'}
            />
          </Col>
        </Row>
      </Grid>
    );
  }

  private toggleNotificationTab = (activeMatchViewTab: MatchInterfaces.MatchViewTabType) => {
    this.setState({
      activeMatchViewTab,
    });
  };
}
