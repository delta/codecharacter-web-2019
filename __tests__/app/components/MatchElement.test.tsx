import { MatchElement } from 'app/components/MatchView/MatchElement';
import { mount } from 'enzyme';
import * as React from 'react';

describe('MatchElement Component', () => {
  const wrapper = mount(
    <MatchElement
      index={1}
      match={{
        playedAt: 'date string',
        userOne: 'user1',
        userOneScore: 1,
        userTwo: 'user2',
        userTwoScore: 5,
        winner: 'user2',
      }}
    />,
  );

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
