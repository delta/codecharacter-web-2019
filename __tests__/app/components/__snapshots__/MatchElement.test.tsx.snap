// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchElement Component Should render 1`] = `
<MatchElement
  index={1}
  match={
    Object {
      "playedAt": "date string",
      "userOne": "user1",
      "userOneScore": 1,
      "userTwo": "user2",
      "userTwoScore": 5,
      "winner": "user2",
    }
  }
>
  <Col
    bsClass="col"
    className="mb-1"
    componentClass="div"
    sm={12}
  >
    <div
      className="mb-1 col-sm-12"
    >
      <div
        className="w-100 undefined"
      >
        <div
          className="d-flex justify-content-around text-dark"
        >
          <span>
            user1
          </span>
          <span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f3ed",
                    "M496 128c0 221.282-135.934 344.645-221.539 380.308a48 48 0 0 1-36.923 0C130.495 463.713 16 326.487 16 128a48 48 0 0 1 29.539-44.308l192-80a48 48 0 0 1 36.923 0l192 80A48 48 0 0 1 496 128zM256 446.313l.066.034c93.735-46.689 172.497-156.308 175.817-307.729L256 65.333v380.98z",
                  ],
                  "iconName": "shield-alt",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-shield-alt fa-w-16 "
                data-icon="shield-alt"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 128c0 221.282-135.934 344.645-221.539 380.308a48 48 0 0 1-36.923 0C130.495 463.713 16 326.487 16 128a48 48 0 0 1 29.539-44.308l192-80a48 48 0 0 1 36.923 0l192 80A48 48 0 0 1 496 128zM256 446.313l.066.034c93.735-46.689 172.497-156.308 175.817-307.729L256 65.333v380.98z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
          <span>
            user2
          </span>
        </div>
        <div
          className=" d-flex justify-content-center text-capitalize text-font-weight-bold text-dark h2  my-2"
        >
          1
           - 
          5
        </div>
        <div
          className=" d-flex justify-content-center text-capitalize text-font-weight-bold text-dark"
        >
          LOST
        </div>
        <div
          className="d-flex justify-content-around text-dark"
        >
          <span
            className=""
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f03d",
                    "M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z",
                  ],
                  "iconName": "video",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-video fa-w-18 "
                data-icon="video"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
          <span
            className=""
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f03d",
                    "M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z",
                  ],
                  "iconName": "video",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-video fa-w-18 "
                data-icon="video"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
          <span
            className=""
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f03d",
                    "M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z",
                  ],
                  "iconName": "video",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-video fa-w-18 "
                data-icon="video"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
          <span
            className=""
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f03d",
                    "M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z",
                  ],
                  "iconName": "video",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-video fa-w-18 "
                data-icon="video"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
          <span
            className=""
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f03d",
                    "M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z",
                  ],
                  "iconName": "video",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-video fa-w-18 "
                data-icon="video"
                data-prefix="fas"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </span>
        </div>
        <div
          className=" d-flex justify-content-start text-capitalize text-dark my-1 mx-3 h6"
          style={
            Object {
              "fontSize": "12px",
            }
          }
        >
          Played At : 
          date string
        </div>
      </div>
    </div>
  </Col>
</MatchElement>
`;
